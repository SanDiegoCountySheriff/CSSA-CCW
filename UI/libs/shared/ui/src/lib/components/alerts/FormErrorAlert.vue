<template>
  <v-alert class="mt-5" type="error" elevation="3" dismissible>
    Can not continue do to errors in the following fields
    <span v-for="(error, i) in errors" :key="i"> {{ error }}, </span>
  </v-alert>
</template>

<script lang="ts">
import { defineComponent } from 'vue';

export default defineComponent({
  name: 'FormErrorAlert',
  props: {
    errors: {
      type: Array<string>,
      default: () => []
    }
  },
});
</script>

