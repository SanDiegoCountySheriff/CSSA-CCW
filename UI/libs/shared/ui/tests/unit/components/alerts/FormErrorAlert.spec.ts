/* eslint-disable @typescript-eslint/ban-ts-comment */
import FormErrorAlert from '@shared-ui/components/alerts/FormErrorAlert.vue';
import Vuetify from 'vuetify';
import { createLocalVue, mount } from '@vue/test-utils';

const localVue = createLocalVue();

describe('FormErrorAlert', () => {
  let vuetify;

  beforeEach(() => {
    vuetify = new Vuetify();
  });

  it('should match the snapshot', () => {
    //@ts-ignore
    const wrapper = mount(FormErrorAlert, {
      localVue,
      vuetify,
      mocks: {
        // eslint-disable-next-line @typescript-eslint/no-empty-function
        $t: () => {},
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  it('Should render the passed in error', () => {
    //@ts-ignore
    const wrapper = mount(FormErrorAlert, {
      localVue,
      vuetify,
      propsData: {
        errors: ['error1'],
      },
      mocks: {
        // eslint-disable-next-line @typescript-eslint/no-empty-function
        $t: () => {},
      },
    });

    expect(wrapper.text()).toContain('error1');
  });

  it('should render multiple errors', () => {
    //@ts-ignore
    const wrapper = mount(FormErrorAlert, {
      localVue,
      vuetify,
      propsData: {
        errors: ['error1', 'error2'],
      },
      mocks: {
        // eslint-disable-next-line @typescript-eslint/no-empty-function
        $t: () => {},
      },
    });

    expect(wrapper.text()).toContain('error1');
    expect(wrapper.text()).toContain('error2');
  });
});
