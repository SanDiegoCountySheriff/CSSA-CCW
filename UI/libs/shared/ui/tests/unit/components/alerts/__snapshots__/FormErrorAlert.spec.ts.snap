// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormErrorAlert should match the snapshot 1`] = `
<div role="alert" class="v-alert mt-5 v-sheet theme--dark elevation-3 error">
  <div class="v-alert__wrapper"><i aria-hidden="true" class="v-icon notranslate v-alert__icon mdi mdi-alert theme--dark"></i>
    <div class="v-alert__content">

    </div><button type="button" class="v-alert__dismissible v-btn v-btn--icon v-btn--round theme--dark v-size--small" aria-label="Close"><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate mdi mdi-close-circle theme--dark"></i></span></button>
  </div>
</div>
`;
